### ISSUE TITLE
FinalParameters should not warn for non-final parameters of native methods 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#2981]
SForge request: 468
 Created: 2007-07-28
Creator: Albert Strasheim
Private: No

With the "FinalParameters" check turned on, the following method declaration:

`private native void foo(int bar);`

generates a warning: Parameter bar should be final

As far as I can figure, having final here serves no useful purpose here. This is similar to the abstract method case, which doesn't generate a warning:

`protected abstract void bar(int baz);`

```
/var/tmp $ javac TestClass.java 
/var/tmp $ cat TestClass.java 
public class TestClass {

	private native void foo(int bar);
}

/var/tmp $ cat Test.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
        <module name="FinalParameters"/>
    </module>
</module>

/var/tmp $ java -jar checkstyle-6.16-SNAPSHOT-all.jar -c Test.xml TestClass.java
Starting audit...
[ERROR] /var/tmp/TestClass.java:3:33: Parameter bar should be final. [FinalParameters]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
fix is merged .
